net/socket.h

```diff
@@ -216,6 +216,23 @@ namespace net {
         // may block once at most, when there's no data yet in the socket;
         virtual ssize_t recv(void *buf, size_t count, int flags = 0) = 0;
         virtual ssize_t recv(const struct iovec *iov, int iovcnt, int flags = 0) = 0;
+        virtual ssize_t recv_mutable(struct iovec *iov, int iovcnt, int flags = 0) {
+            return recv(iov, iovcnt, flags);
+        }
```

Comment:
Because ```const iovec*``` is a convention, just like read/readv/pread/preadv/write/writev/pwrite/pwritev ...

Obeying the convention can avoid unnecessary errors.

Reply:
Using another function name is also for obeying the convention, despite C++ has function overloading.

Reply:
Oh, I made a mistake. In fact recv/send usually makes a single I/O action that doesn't need to modify the iovec[]. Only recv_at_least needs so.